src/main/java/com/google/devtools/build/lib/skyframe/AspectFunction.java

```diff
@@ -186,6 +187,15 @@ public SkyValue compute(SkyKey skyKey, Environment env)
     ConfiguredTarget associatedTarget = state.initialValues.associatedTarget;
     Target target = state.initialValues.target;
 
+    if (associatedTarget.get(IncompatiblePlatformProvider.PROVIDER) != null) {
```

Comment:
Sanity check: this just means that we return an aspect that does nothing?

Reply:
That's my understanding, yes.

And more importantly, as far as I can tell, it doesn't trigger any dependent aspects from getting evaluated. Which should address the question you had I think. The one about what happens if an incompatible target depends on a compatible one. The aspect does _not_ get evaluated for the compatible dependencies.

Reply:
Oh, I honestly can't remember where aspect propagation is implemented (AspectFunction vs. ConfiguredTargetFunction vs. DependencyResolver vs ...).

But thanks to your test coverage we have good formalized behavior expectations, so my question doesn't matter!

Thanks for verifying.